File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        for(int i = 0; i<nums.size(); i++){
            int num = nums[i];
            for(int j = i+1;j<nums.size();j++){
                if(nums[j] == num){
                    return true;
                }
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefix_sum = {0: 1}
        curr_sum = 0
        count = 0

        for num in nums:
            curr_sum += num
            if curr_sum - k in prefix_sum:
                count += prefix_sum[curr_sum - k]
            prefix_sum[curr_sum] = prefix_sum.get(curr_sum, 0) + 1
        return count